### PR TITLE
Coverity fixes

### PR DESCRIPTION
**Pull request by nigels-com**
See also: https://scan.coverity.com/projects/nigels-com-displaz

### CODE REVIEW

File: src/render/PointArray.cpp [hunk 1]
```diff
@@ -154,8 +154,8 @@ static OctreeNode* makeTree(int depth, size_t* inds,
         return node;
     }
     // Partition points into the 8 child nodes
-    size_t* childRanges[9];
-    multi_partition(beginPtr, endPtr, OctreeChildIdx(P, center), childRanges+1, 8);
+    std::vector<size_t *> childRanges(9, NULL);
```
Comment (c42f) [inline src/render/PointArray.cpp]: I assume this is about initialization, can we get away with `size_t* childRanges[9] = {0};` instead?  Seems a bit unfortunate to hit the allocator for this small fixed size array.

File: src/main.cpp [hunk 2]
```diff
@@ -248,10 +248,16 @@ int main(int argc, char *argv[])
...
-        std::cout << "\n";
+        try
+        {
+            channel->sendMessage("QUERY_CURSOR");
+            QByteArray msg = channel->receiveMessage();
+            std::cout.write(msg.data(), msg.length());
+            std::cout << "\n";
+        }
+        catch (DisplazError & e) // receiveMessage can time out
+        {
```
Comment (c42f) [inline src/main.cpp]: Looks good.  Let's also log the error to `std::cerr` here, to give the user some sort of feedback.

File: src/render/PointArray.cpp (hunk 1 above)
Comment (nigels-com) [inline src/render/PointArray.cpp]: Yes, that would work.  `std::array` is an option too, assuming C++11

```
std::array<size_t *, 9> childRanges;
childRanges.fill(NULL);
````

which is arguably a bit clearer than the alternatives.

Comment (c42f) [inline src/render/PointArray.cpp]: Should be safe to assume std::array is available.  Personally I'm perfectly happy with a good old C array, provided you don't need the type preserved as it's passed around.

**Comment by nigels-com** [comment]
Updated, ought to be good to go.

**Comment by c42f** [comment]
Cheers, thanks

**Pull request merged**